cypress/e2e/swap.test.ts

```diff
@@ -75,35 +77,88 @@ describe('Swap', () => {
   })
 
   it('can swap ETH for USDC', () => {
-    const TOKEN_ADDRESS = USDC_MAINNET.address
     const BALANCE_INCREMENT = 1
-    cy.hardhat().then((hardhat) => {
-      cy.then(() => hardhat.getBalance(hardhat.wallet.address, USDC_MAINNET))
-        .then((balance) => Number(balance.toFixed(1)))
-        .then((initialBalance) => {
-          cy.get('#swap-currency-output .open-currency-select-button').click()
-          cy.get(getTestSelector('token-search-input')).clear().type(TOKEN_ADDRESS)
-          cy.contains('USDC').click()
-          cy.get('#swap-currency-output .token-amount-input').clear().type(BALANCE_INCREMENT.toString())
-          cy.get('#swap-currency-input .token-amount-input').should('not.equal', '')
-          cy.get('#swap-button').click()
-          cy.get('#confirm-swap-or-send').click()
-          cy.get(getTestSelector('dismiss-tx-confirmation')).click()
-
-          cy.then(() => hardhat.provider.send('hardhat_mine', ['0x1', '0xc'])).then(() => {
-            // ui check
+    cy.visit('/swap', { ethereum: 'hardhat' })
+      .hardhat()
+      .then((hardhat) => {
+        hardhat
+          .reset()
+          .then(() => hardhat.provider.send('evm_setAutomine', [true]))
```

Comment:
Doesn't this need to be backed out after being set? How do you know it won't affect the subsequent test?

Reply:
I think in this case since we're planning on having automine set to true by default it's just a defensive safety check

Reply:
If anything I think it could be removed (on the assumption that *other* tests which set it to false unwind that by setting it to true at the end.